RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -0,0 +1,202 @@
+        }
+    }
+
+    var landscapeRightComponents: [PaywallComponent] {
+        return self.components.filter { component in
+            guard let displayPreferences = component.displayPreferences else {
+                return true
+            }
+            return displayPreferences.contains(.landscapeRight)
+        }
+    }
+
+    var portraitComponents: [PaywallComponent] {
+        return self.components.filter { component in
+            guard let displayPreferences = component.displayPreferences else {
+                return true
+            }
+            return displayPreferences.contains(.portrait)
+        }
+    }
+
+    var body: some View {
+        if self.shouldUseLandscapeLayout {
+            HStack {
+                VStack(spacing: 0) {
+                    self.layoutComponents(self.landscapeLeftComponents)
+                }
+//                .scrollableIfNecessaryWhenAvailable()
+
+                VStack(spacing: 0) {
+                    self.layoutComponents(self.landscapeRightComponents)
+                }
+//                .scrollableIfNecessaryWhenAvailable()
+            }
+
+        } else {
+            self.layoutComponents(self.portraitComponents)
+//                .scrollableIfNecessaryWhenAvailable()
+        }
+    }
+
+    @ViewBuilder
+    func layoutComponents(_ layoutComponents: [PaywallComponent]) -> some View {
+        ForEach(Array(layoutComponents.enumerated()), id: \.offset) { index, item in
+            switch (item) {
+            case .tiers(let component):
+                TiersComponentView(
+                    locale: locale,
+                    component: component,
+                    configuration: configuration
+                )
+            case .tierSelector:
+                // This gets displayed in TiersComponentView right now
+                EmptyView()
+            case .tierToggle:
+                // This gets displayed in TiersComponentView right now
+                EmptyView()
+            case .text(let component):
+                TextComponentView(locale: locale, component: component)
+            case .image(let component):
+                ImageComponentView(locale: locale, component: component)
+            case .spacer(let component):
+                SpacerComponentView(
+                    locale: locale,
+                    component: component
+                )
+            }
+        }
+    }
+
+    @Environment(\.userInterfaceIdiom)
+    var userInterfaceIdiom
+
+    var defaultHorizontalPaddingLength: CGFloat? {
+        return Constants.defaultHorizontalPaddingLength(self.userInterfaceIdiom)
+    }
+
+    var defaultVerticalPaddingLength: CGFloat? {
+        return Constants.defaultVerticalPaddingLength(self.userInterfaceIdiom)
+    }
+
+    #if swift(>=5.9) || (!os(macOS) && !os(watchOS) && !os(tvOS))
+    @Environment(\.verticalSizeClass)
+    var verticalSizeClass
+    #endif
+
+    var shouldUseLandscapeLayout: Bool {
```

Comment:
seems like we're only using this variable here, can it be made private?

Reply:
Took this out.